### PR TITLE
Abstract FaceCameraSource

### PR DESCRIPTION
**Pull request by alexstyl**
Currently the only way of getting the user 'mood' is through the camera. This is a problematic for debugging purposes as you need to act like a clown in order to debug the app 😂 

This PR abstracts the `FaceReactionSource` so that we can swap out different 'sources' depending the need.

On a follow up PR: Allow fake face detection from the keyboard ⌨️ (https://github.com/danybony/magic-mirror-dashboard/pull/8)

### CODE REVIEW

File: app/src/main/java/net/bonysoft/magicmirror/facerecognition/FaceCameraSource.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Override
+    public void start() {
+        try {
+            preview.start(cameraSource);
+        } catch (IOException e) {
+            Log.e(e, "Unable to start camera source.");
+            cameraSource.release();
+            cameraSource = null;
```
Comment (danybony) [inline app/src/main/java/net/bonysoft/magicmirror/facerecognition/FaceCameraSource.java]: It is not needed to nullify this anymore

File: app/src/main/java/net/bonysoft/magicmirror/facerecognition/FaceCameraSource.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        }
+    }
+
+    @Override
+    public void release() {
+        cameraSource.release();
+        cameraSource = null;
```
Comment (danybony) [inline app/src/main/java/net/bonysoft/magicmirror/facerecognition/FaceCameraSource.java]: Same here, this should not be needed anymore

**Comment by alexstyl** [comment]
@danybony comments addressed in https://github.com/danybony/magic-mirror-dashboard/pull/7/commits/7caa66958e02dbcc5d15ba5f74c36a6083812ec3

**Comment by danybony** [comment]
👍

**Pull request merged**